### PR TITLE
Fixed issue #18343: Problem with interpreting double slash in URL (IIS)

### PR DESCRIPTION
**Pull request by Shnoulle**
Fixed issue #18747: user administration email - reserved words send 403 error

### CODE REVIEW

**Comment by Shnoulle** [comment]
Double slash is needed to don't throw 400 error with empty action.
Else : no control of permisison on survey id, need globalsettings permision ?

Return an empty array …

**Comment by Shnoulle** [comment]
@gabrieljenik alreadey reviewed on https://github.com/LimeSurvey/LimeSurvey/pull/2630 (with your advice) and https://github.com/LimeSurvey/LimeSurvey/pull/2639 

Added the test on surveyid here

File: application/helpers/admin/htmleditor_helper.php [hunk 1]
```diff
@@ -208,12 +208,91 @@ function getPopupEditor($fieldtype, $fieldname, $fieldtext, $surveyID = null, $g
...
         $class = "editorLink input-group-text";
     }
-    $htmlcode .= ""
-    . "<a href=\"javascript:start_popup_editor('" . $fieldname . "','" . addslashes(htmlspecialchars_decode((string) $fieldtext, ENT_QUOTES)) . "','" . $surveyID . "','" . $gID . "','" . $qID . "','" . $fieldtype . "','" . $action . "')\" id='" . $fieldname . "_ctrl' class='{$class} btn btn-outline-secondary btn-xs'>\n"
-    . "\t<i class='ri-pencil-fill btneditanswerena' id='" . $fieldname . "_popupctrlena' data-bs-toggle='tooltip' data-bs-placement='bottom' title='" . gT("Start HTML editor in a popup window") . "'></i>"
-    . "\t<i class='ri-pencil-fill btneditanswerdis' id='" . $fieldname . "_popupctrldis'  style='display:none'  ></i>"
-    . "</a>\n";
-
+    /* @var string[] parameters of the editor url */
+    $editorurlparams = array(
```
Comment (c-schmitz) [inline application/helpers/admin/htmleditor_helper.php:212]: For new variables we decided for lowerCamelCase - please check out https://manual.limesurvey.org/Code_quality_guide#Variables

**Review by c-schmitz — CHANGES_REQUESTED** [review]

File: application/helpers/admin/htmleditor_helper.php (hunk 1 above)
Comment (Shnoulle) [reply application/helpers/admin/htmleditor_helper.php:212]: Copy code from september 2022 …

Only in 6.X ? Right ?

**Comment by Shnoulle** [comment]
On JS file too ?

**Comment by Shnoulle** [comment]
Better now, don't remove other 6.X fixes …

File: application/controllers/LimereplacementfieldsController.php [hunk 2]
```diff
@@ -37,7 +37,7 @@ public function actionIndex()
             throw new CHttpException(401);
         }
 
-        if (!Permission::model()->hasSurveyPermission($surveyid, 'survey', 'read')) {
+        if ($surveyid && !Permission::model()->hasSurveyPermission($surveyid, 'survey', 'read')) {
```
Comment (gabrieljenik) [inline application/controllers/LimereplacementfieldsController.php:40]: Shouldn't we check this and throw exception prevously, maybe before querying the survye model?

Comment (Shnoulle) [reply application/controllers/LimereplacementfieldsController.php:40]: No it was added to fix a security issue

It's your commit
https://github.com/LimeSurvey/LimeSurvey/commit/64e2003319ffffdcd91d98cb22d9c48f5e86782b

Comment (Shnoulle) [reply application/controllers/LimereplacementfieldsController.php:40]: PS : must not be surveycontent/read here ?

Comment (gabrieljenik) [reply application/controllers/LimereplacementfieldsController.php:40]: I was referring to checking the surveyid is not empty. Sorry.

Comment (gabrieljenik) [reply application/controllers/LimereplacementfieldsController.php:40]: > PS : must not be surveycontent/read here ?

Yes, think it is a good idea

Comment (Shnoulle) [reply application/controllers/LimereplacementfieldsController.php:40]: > I was referring to checking the surveyid is not empty. Sorry.

If surveyid != "" and \Survey not found : must throw a 404 here

But : must throw a 403 only is surveyid != ""  .

If sureyid is not set : return an empty array.

Comment (Shnoulle) [reply application/controllers/LimereplacementfieldsController.php:40]: > Yes, think it is a good idea

Not related to this fix ;)

**Pull request merged**